Assets/MRTK/SDK/Features/Utilities/Migration/Tools/MigrationTool.cs

```diff
@@ -362,6 +418,34 @@ private static bool IsSceneAsset(Object selectedObject)
         {
             return selectedObject is SceneAsset;
         }
+
+        /// <summary>
+        /// Utility class to keep migration status of each object
+        /// </summary>
+        public class MigrationStatus
+        {
+            /// <summary>
+            /// Flag to indicate if object was already processed by migration
+            /// </summary>
+            public bool isProcessed { get; set; }
+
+            /// <summary>
+            /// Keep track of the amount of issues found during migration process of every children object in the migration object hierarchy
+            /// </summary>
+            public int failures { get; set; }
+
+            /// <summary>
+            /// Keep track of recorded messages logged during  the migration process
+            /// </summary>
+            public String log { get; set; }
```

Comment:
Nit: From a class design perspective, if a consumer is going to be adding messages to a log, it might be better to wrap that within this (i.e. addToLog(String text) and then outputToLog()) or something.

In this case it's pretty minor because usage of this as a simple place to put stuff that survives a loop is maybe fine for what you're doing here